cli_tools/daisy_test_runner/main.go

```diff
@@ -554,13 +554,16 @@ func runTestCase(ctx context.Context, test *test, tc *junitTestCase, errors chan
 
 	start := time.Now()
 	fmt.Printf("[TestRunner] Running test case %q\n", tc.Name)
-	if err := test.testCase.w.Run(ctx); err != nil {
+	err = test.testCase.w.Run(ctx)
+	tc.Time = time.Since(start).Seconds()
+	tc.SystemOut = test.testCase.logger.buf.String()
```

Comment:
minor: this seems like a violation of the [Law of Demeter](https://en.wikipedia.org/wiki/Law_of_Demeter). Not a big deal but it can lead to code being brittle and having to change anytime you refactor or update the implementation. Consider adding a function so this can be something like `test.testCase.getLogBuffer()` or something. But I also see this is just a refactor so your call.